opal/mca/btl/ofi/btl_ofi_component.c

```diff
@@ -581,21 +581,25 @@ static int mca_btl_ofi_init_device(struct fi_info *info)
 fail:
     /* clean up */
 
+    if (NULL != ep && !module->is_scalable_ep) {
+        fi_close(&ep->fid);
+    }
+
     /* if the contexts have not been initiated, num_contexts should
      * be zero and we skip this. */
     for (int i=0; i < module->num_contexts; i++) {
         mca_btl_ofi_context_finalize(&module->contexts[i], module->is_scalable_ep);
     }
     free(module->contexts);
 
-    if (NULL != av) {
-        fi_close(&av->fid);
-    }
-
     if (NULL != ep) {
         fi_close(&ep->fid);
     }
 
+    if (NULL != av) {
+        fi_close(&av->fid);
```

Comment:
Seems like there's an ordering necessity here. It would be good to comment why this needs to come after freeing `ep->fid`

Reply:
Thank you. I've added a brief comment.